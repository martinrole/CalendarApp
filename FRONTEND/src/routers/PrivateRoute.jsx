import { Routes, Route, Navigate } from "react-router-dom";
import { CalendarScreen } from "../components/calendar/CalendarScreen";


export const PrivateRoute = () => {
    console.log('PrivateRoute');

    return (
        <Routes>
            <Route path="/calendar" element={<CalendarScreen />} />
            <Route path="/*" element={<Navigate to="/calendar" />} />
        </Routes>
    )
}
