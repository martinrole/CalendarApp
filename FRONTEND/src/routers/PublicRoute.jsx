import { Routes, Route } from "react-router-dom";
import { NotFoundScreen } from '../components/NotFound/NotFoundScreen'
import { LoginScreen } from '../components/auth/LoginScreen'

export const PublicRoute = () => {
    console.log('PublicRoute');

    return (
        <Routes>
            <Route path="/login" element={<LoginScreen />} />
            <Route path="/*" element={<NotFoundScreen />} />
        </Routes>

    )
}