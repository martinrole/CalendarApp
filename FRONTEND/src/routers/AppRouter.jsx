import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { Loader } from "../components/ui/Loader";
import { authCheckAction } from "../redux/ducks/authDucks";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {

    const { checking, uid } = useSelector(state => state.auth)

    const dispatch = useDispatch()

    useEffect(() => {
        //Valida que el usuario esté autenticado
        dispatch(authCheckAction())
    }, [dispatch])

    if (checking) {
        return <Loader />
    }

    return (
        <BrowserRouter>
            <Routes>
                {uid && <Route path="/*" element={<PrivateRoute />} />}
                {!uid && <Route path="/*" element={<PublicRoute />} />}
            </Routes>
        </BrowserRouter>
    )
}


//Información importante sobre react-router-dom V6 
//https://reactrouter.com/docs/en/v6/upgrading/v5

//Ajustar Redirect:
//https://gist.github.com/mjackson/b5748add2795ce7448a366ae8f8ae3bb
//https://stackoverflow.com/questions/67050966/how-to-build-a-404-page-with-react-router-dom-v6

