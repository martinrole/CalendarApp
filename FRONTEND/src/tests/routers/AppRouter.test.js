import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';
import '@testing-library/jest-dom'
import { AppRouter } from '../../routers/AppRouter';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('Pruebas en <AppRouter />', () => {
    
    test('Debe mostrar el Loading', () => {

        const initialState = {
            auth: {
                checking: true
            }
        }

        const store = mockStore(initialState)

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        )
        
        expect(wrapper).toMatchSnapshot()
        expect( wrapper.find('span').exists() ).toBe(true)
    });

    test('Debe mostrar ruta pÃºblica', () => {

        const initialState = {
            auth: {
                checking: false,
                uid: null
            }
        }

        const store = mockStore(initialState)

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        )
        
        expect(wrapper).toMatchSnapshot()
        expect( wrapper.find('h2').exists() ).toBe(true)
    });

    test('Debe mostrar ruta privada', () => {

        const initialState = {
            ui: {
                modalOpen: false
            },
            calendar: {
                events: []
            },
            auth: {
                checking: false,
                uid: 'abc1234'
            }
        }

        const store = mockStore(initialState)

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        )
        
        //expect(wrapper).toMatchSnapshot()
        expect( wrapper.find('.calendar-screen').exists() ).toBe(true)
    });



})
