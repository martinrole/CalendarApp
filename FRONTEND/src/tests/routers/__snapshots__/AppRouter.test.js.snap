// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe mostrar el Loading 1`] = `
<div
  className="d-flex justify-content-center align-items-center"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="spinner-grow text-primary ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-secondary ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-success ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-danger ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-warning ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-info ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-light ml-3"
    role="status"
  />
  <div
    className="spinner-grow text-dark ml-3"
    role="status"
  />
  <span
    className="sr-only"
  >
    Loading...
  </span>
</div>
`;

exports[`Pruebas en <AppRouter /> Debe mostrar ruta p√∫blica 1`] = `
<div
  className="d-flex justify-content-center align-items-center"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div>
    <h1
      className="text-center text-danger"
    >
      404
    </h1>
    <h2
      className="mb-4"
    >
      Ops! Page Not Found
    </h2>
    <button
      className="btn btn-primary btn-block"
      onClick={[Function]}
    >
      Ingresar
    </button>
  </div>
</div>
`;
